app/helpers/hearing_paginator.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+# Creates a chronologically sorted set of "items"
+# with next and previous navigation helpers.
+#
+# Next and previous correlate to navigation over all
+# a cases' hearing's hearing days.
+#
+class HearingPaginator
+  include ActionView::Helpers
+  include Rails.application.routes.url_helpers
+
+  PageItem = Struct.new(:id, :hearing_date)
+
+  # rubocop:disable Rails/HelperInstanceVariable
+  def initialize(prosecution_case, options = {})
+    @prosecution_case = prosecution_case
+    @current_page = options.fetch(:page, nil).to_i
+  end
+  # rubocop:enable Rails/HelperInstanceVariable
+
+  def items
+    sorted_hearing_page_items
+  end
+
+  def current_item
+    items[current_page || first_page]
+  end
+
+  attr_writer :current_page
+
+  def current_page
+    @current_page ||= 0
+  end
+
+  def first_page
+    0
+  end
+
+  def last_page
+    [items.size - 1, 0].max
+  end
+
+  def first_page?
+    current_page == first_page
+  end
+
+  def last_page?
+    current_page == last_page
+  end
+
+  def next_page_link
+    link_to(t('hearings.show.pagination.next_page'),
+            hearing_path(id: next_item.id,
+                         urn: prosecution_case.prosecution_case_reference,
+                         page: next_page),
+            class: 'govuk-link app-pagination-next')
+  end
+
+  def previous_page_link
+    link_to(t('hearings.show.pagination.previous_page'),
+            hearing_path(id: previous_item.id,
+                         urn: prosecution_case.prosecution_case_reference,
+                         page: previous_page),
+            class: 'govuk-link app-pagination-previous')
+  end
+
+  private
+
+  def next_item
+    items[next_page]
+  end
+
+  def next_page
+    [current_page + 1, last_page].min
+  end
+
+  def previous_item
+    items[previous_page]
+  end
+
+  def previous_page
+    [current_page - 1, first_page].max
+  end
+
+  def sorted_hearing_page_items
+    hearing_page_items.sort_by(&:hearing_date)
```

Comment:
Having pulled this branch and tested, the sorted_hearing_page_items order here does not necessarily match the order of the hearing dates in the Hearings table, if the dates of the Hearings overlap (possibly an edge case???). This means that when paging through hearing days the order doesn't necessarily match the order in the Hearings table (which I think DF said was a requirement) but also (more seriously) means that when you click on a date in the Hearings table you could get taken to the wrong day. As mentioned above I think this is probably only an issue when the hearing days for 2 hearings overlap.

Reply:
I will need to look at this. I assume you have duplicated the problem @kmahern?! I am not sure how edge case it is. It could be caused by differences between `#hearings_with_day_by_datetime` and `#sorted_hearing_page_items` ordering.

Reply:
@kmahern I have [pushed this commit](https://github.com/ministryofjustice/laa-court-data-ui/pull/475/commits/a4fa57807ecdaa7b73fb348ee863cd15cead5fb1)  that should duplicate the ordering between
the case details "hearing summaries" table and the hearing details page pagination ordering. 

This should deal with the difference you encountered so it would be good if can you check this again.

Reply:
👍 Have retested and lgtm.